flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -140,6 +143,7 @@
 
 	private final Time slotRequestTimeout;
 
+	private final Counter numberOfRestartsCounter;
```

Comment:
What is the benefit of using a `Counter` over a `Gauge`? We never decrement the number of restarts.

Reply:
That's a bit of a can of worms.

In my mind, the different metric types are purely syntactic sugar, offering convenience APIs for modifying/retrieving data. As in, conceptually, everything is a gauge.
However, several reporters map metrics based on their class to types that the external system supports. That very metric type is used by certain systems to derive what kind of operations/features are applicable for a given metric.
Thus, a gauge (due to being the most general metric type) can potentially be less useful to a user than a counter.

We haven't truly reached a conclusive stance on the matter; ultimately we should either force all reporters to report metrics as if they were gauges, or decouple the interface (gauge/metric/histogram) from the semantic (arbitrary value, value that only increments, rates).

Reply:
If we can have the conclusion to not support the meter, I'd prefer to use a gauge which can make the change simpler. If a meter `numberOfRestartsPerSecond` need to be introduced, a counter will be simpler since it can be the underlying data source of a MeterView. 
A gauge cannot be used by a meter since meter has write interfaces `marEvent(...)` while `Gauge` is readonly. An alternative is to introduce a ReadOnlyMeterView which does not allow invocation of `marEvent(...)`.

Reply:
We should use a Gauge. If a meter is needed later, we could implement a `ReadOnlyMeterView` that takes a Gauge, as you said.

Reply:
Ok. done.